saleor/events/models/OrderEvent.py

```diff
@@ -0,0 +1,274 @@
+from typing import Dict, List, Optional, Tuple, Union
+
+from django.conf import settings
+from django.contrib.auth.base_user import AbstractBaseUser
+from django.contrib.postgres.fields import JSONField
+from django.db import models
+from django.utils.timezone import now
+from prices import Money
+
+from ...account.models import Address
+from ...core.utils.json_serializer import CustomJsonEncoder
+from ...order.models import Fulfillment, Order, OrderLine
+from ...payment.models import Payment
+from ...product.models import ProductVariant
+from ..lang.order_events import display_order_event
+from ..types import OrderEvents, OrderEventsEmails
+
+User = AbstractBaseUser
+
+
+class OrderEvent(models.Model):
+    """Model used to store events that happened during the order lifecycle.
+
+        Args:
+            parameters: Values needed to display the event on the storefront
+            type: Type of an order
+    """
+    date = models.DateTimeField(default=now, editable=False)
+    type = models.CharField(
+        max_length=255,
+        choices=((event.name, event.value) for event in OrderEvents))
+    order = models.ForeignKey(
+        Order, related_name='events', on_delete=models.CASCADE)
+    parameters = JSONField(
+        blank=True, default=dict, encoder=CustomJsonEncoder)
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL, blank=True, null=True,
+        on_delete=models.SET_NULL, related_name='+')
+
+    class Meta:
+        ordering = ('date', )
+        db_table = 'order_orderevent'
+
+    def __repr__(self):
+        return 'OrderEvent(type=%r, user=%r)' % (self.type, self.user)
+
+    @staticmethod
+    def _lines_per_quantity_to_str_line_list(quantities_per_order_line):
+        return [{
+            'quantity': quantity,
+            'item': str(line)
+        } for quantity, line in quantities_per_order_line]
+
+    def get_event_display(self):
+        return display_order_event(self)
+
+    @classmethod
+    def email_sent_event(
+            cls, *,
+            order: Order, email_type: OrderEventsEmails,
+            source: Optional[User]) -> models.Model:
```

Comment:
I would consider renaming all `source` arguments to `user`, it would be much more readable.